### PR TITLE
Fix RessourceWarnings in ImageFileCollection._generator

### PR DESCRIPTION
**Pull request by MSeifert04**
- [x] Did you add an entry to the "Changes.rst" file?
- [ ] Did you add a regression test?
- [ ] Does the commit message include a "Fixes #issue_number" (replace "issue_number").
- [ ] Does this PR add, rename, move or remove any existing functions or parameters?

Supercedes #545 which I messed up (without having a backup branch 😅 )

### CODE REVIEW

File: ccdproc/image_collection.py [hunk 1]
```diff
@@ -820,29 +820,34 @@ def _generator(self, return_type,
...
+                return_thing = fits.getdata(full_path, self.ext, **add_kwargs)
+            elif return_type == 'ccd':
+                return_thing = fits_ccddata_reader(
+                    full_path, hdu=self.ext, **ccd_kwargs)
+            elif return_type == 'hdu':
+                with fits.open(full_path, **add_kwargs) as hdulist:
+                    ext_index = hdulist.index_of(self.ext)
+                    # Need to copy the HDU to prevent lazy loading problems
+                    # and "IO operations on closed file" errors
+                    return_thing = hdulist[ext_index].copy()
```
Comment (MSeifert04) [inline ccdproc/image_collection.py]: Unfortunately this `.copy()` was the cleanest way I found to make it not throw "fileio on closed file". But if you have another suggestion that would be fine!

**Comment by MSeifert04** [comment]
@mwcraig @crawfordsm Does this look okay?

File: ccdproc/image_collection.py (hunk 1 above)
Comment (mwcraig) [reply ccdproc/image_collection.py]: This is probably necessary because of lazy loading of FITS files....

Comment (MSeifert04) [reply ccdproc/image_collection.py]: Not sure. I also suspect `memmap`. However after I realized that `.copy()` does the trick I stopped further investigation. I was just interested if you know a better way.

**Pull request merged**